lib/stanford-mods/date_parsing.rb

```diff
@@ -85,6 +86,55 @@ def self.facet_string_for_century(date_str)
         end
       end
 
+      # get String sortable value for B.C. if we have  B.C. pattern
+      #  note that these values must *lexically* sort to create a chronological sort.
+      #  We know our data does not contain B.C. dates older than 999, so we can make them
+      #  lexically sort by subtracting 1000.  So we get:
+      #    -700 for 300 B.C., -750 for 250 B.C., -800 for 200 B.C., -801 for 199 B.C.
+      # @param [String] date_str String containing B.C.
+      # @return [String, nil] String sortable -ddd if B.C. in pattern; nil otherwise
+      def self.sortable_year_for_bc(date_str)
+        bc_matches = date_str.match(/(\d{1,4}).*B\.C\./) if date_str
+        return ($1.to_i - 1000).to_s if bc_matches
+      end
+
+      # get single facet value for B.C. if we have  B.C. pattern
+      # @param [String] date_str String containing B.C.
+      # @return [String, nil] ddd B.C.  if ddd B.C. in pattern; nil otherwise
+      def self.facet_string_for_bc(date_str)
+        bc_matches = date_str.match(/\d{1,4}.*B\.C\./) if date_str
```

Comment:
IMO having a method with a name that describes what it is we're checking is clearer than having to mentally parse the regex (or realize it's the same regex as above once I've read half of it).  At least then, once I have mentally come to terms with what it is that method checks, I don't have to re-parse when I see a reference to that method again.  (I may just be a DRY stickler though)

FWIW, that makes a lot more sense if we are using an instance of the class so we could be more declarative like just saying `return unless bc_date_match?` or something like that.